src/ucp/tag/tag_match.c

```diff
@@ -85,6 +85,41 @@ int ucp_tag_exp_remove(ucp_tag_match_t *tm, ucp_request_t *req)
     return 0;
 }
 
+static void
+ucp_tag_request_exp_queue_erase(ucp_tag_match_t *tm, ucp_request_queue_t *queue,
+                                ucp_tag_t tag, ucp_tag_t mask)
+{
+    ucs_queue_iter_t iter;
+    ucp_request_t *req;
+
+    ucs_queue_for_each_safe(req, iter, &queue->queue, recv.queue) {
+        if (!ucp_tag_is_match(tag, req->recv.tag.tag,
+                              mask & req->recv.tag.tag_mask)) {
```

Comment:
maybe use just "cancel_mask"? Otherwise would be hard to predict what is cancelled by this new func

Reply:
I think recv tag mask need to take into account as well, otherwise we can have meaningless bits in recv tag which were "masked" on posting recv, so such requests won't be canceled.

Reply:
but do you really want to cancel requests posted with less strict mask than you specified?
Can accidentally cancel wrong requests then

Reply:
I don't think these are wrong if user disables some bits from tag by mask for tag matching engine, it's his intention, is not it?
New API should be able to match and cancel any requests, overwise we can have posted requests which won't match any condition due to masked bits on recv.